storage.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	// "bufio"
+	"context"
+	"log"
+	"net"
+	"os"
+	"strings"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+var dbPool *pgxpool.Pool
+
+func initDb() {
+	ctx := context.Background()
+
+	var err error
+	dbPool, err = pgxpool.Connect(ctx, os.Getenv("DATABASE_URL"))
+	checkError("unable to connect to the database", err)
+
+	schemaBytes, err := os.ReadFile("sql/schema.sql")
+	checkError("error reading `sql/schema.sql`", err)
+
+	_, err = dbPool.Exec(ctx, string(schemaBytes))
```

Comment:
What if the DB was already set up?

Reply:
Thank you for the comment! In `sql/schema.sql` I make sure with `IF NOT EXISTS` that it can be run idempotently. So this should be safe if we need to restart the service or start up two or more replicas at once.

However, I would love to learn if there is a nicer way to do this!